pkg/plugin/restore_pvc_action_plugin.go

```diff
@@ -139,6 +139,22 @@ func (p *NewPVCRestoreItemAction) Execute(input *velero.RestoreItemActionExecute
 		return nil, errors.WithStack(err)
 	}
 
+	// Retrieve storage class mapping information and update pvc StorageClassName with new name
+	storageClassMapping := make(map[string]string)
+	p.Log.Info("Retrieving storage class mapping information from configMap")
+	err = pluginItem.RetrieveStorageClassMapping(restConfig, veleroNs, storageClassMapping, p.Log)
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to retrieve storage class mapping information : %v", err)
+		p.Log.WithError(err).Error(errMsg)
+		return nil, errors.New(errMsg)
+	}
+	p.Log.Info("Updating target PVC storage class based on the storage class mapping")
+	itemSnapshot, err = pluginItem.UpdateSnapshotWithNewStorageClass(restConfig, &itemSnapshot, storageClassMapping, p.Log)
```

Comment:
Add a check here to see if there's any storageClassMapping.  If empty, no need to call UpdateSnapshotWithNewStorageClass.